### PR TITLE
SCAL-151356 Added ExecuteTML as a global method

### PR DESCRIPTION
**Pull request by adityamittal3107**

### CODE REVIEW

File: src/embed/base.ts [hunk 1]
```diff
@@ -109,6 +110,35 @@ export const prefetch = (url?: string, prefetchFeatures?: PrefetchFeatures[]): v
     }
 };
 
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export const executeTML = (data: any): any => {
```
Comment (girish-thoughtspot) [inline src/embed/base.ts:114]: Please add types for the param, since we already know the params data type for api.

File: src/embed/base.ts [hunk 2]
```diff
@@ -109,6 +110,35 @@ export const prefetch = (url?: string, prefetchFeatures?: PrefetchFeatures[]): v
...
+    const { thoughtSpotHost } = config;
+    const payload = {
+        metadata_tmls: data.metadata_tmls,
+        import_policy: data.import_policy || 'PARTIAL',
+        create_new: data.create_new || false,
+    };
+    return fetch(`${thoughtSpotHost}${EndPoints.EXECUTE_TML}`, {
+        method: 'POST',
+        headers: {
```
Comment (girish-thoughtspot) [inline src/embed/base.ts:123]: Check for Auth type, incase of TrustedAuthCookieless, we need to add `Authorization: Bearer ${token}` to the header

File: src/embed/base.ts [hunk 3]
```diff
@@ -109,6 +110,35 @@ export const prefetch = (url?: string, prefetchFeatures?: PrefetchFeatures[]): v
...
 
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export const executeTML = (data: any): any => {
+    const { thoughtSpotHost } = config;
+    const payload = {
+        metadata_tmls: data.metadata_tmls,
+        import_policy: data.import_policy || 'PARTIAL',
+        create_new: data.create_new || false,
+    };
+    return fetch(`${thoughtSpotHost}${EndPoints.EXECUTE_TML}`, {
```
Comment (girish-thoughtspot) [inline src/embed/base.ts:121]: If we are returning a promise, we should resolve it with response or reject with errors. Currently user would not get the response, including GUID, from this method, same for errors.

File: src/embed/base.ts (hunk 1 above)
Comment (adityamittal3107) [reply src/embed/base.ts:114]: Done

File: src/embed/base.ts (hunk 3 above)
Comment (adityamittal3107) [reply src/embed/base.ts:121]: Done

File: src/embed/base.ts (hunk 2 above)
Comment (adityamittal3107) [reply src/embed/base.ts:123]: Done

Comment (adityamittal3107) [reply src/embed/base.ts:123]: Done.

File: src/embed/base.ts [hunk 4]
```diff
@@ -109,6 +129,87 @@ export const prefetch = (url?: string, prefetchFeatures?: PrefetchFeatures[]): v
     }
 };
 
+export const executeTML = async (data: executeTMLInput): Promise<any> => {
+    const { thoughtSpotHost, authType } = config;
+    let authToken = '';
```
Comment (girish-thoughtspot) [inline src/embed/base.ts:134]: Existing sanity function can be used for validating that this is not called before `init` function, with something like:
```
    try {
        sanity(embedConfig);
    } catch (err) {
        return Promise.reject(err)
    }

File: src/embed/base.ts [hunk 5]
```diff
@@ -109,6 +129,87 @@ export const prefetch = (url?: string, prefetchFeatures?: PrefetchFeatures[]): v
...
+    })
+        .then((response) => {
+            if (!response.ok) {
+                throw new Error(`Failed to import TML data: ${response.status} - ${response.statusText}`);
+            }
+            return response.json();
+        })
+        .catch((error) => {
+            console.error(error);
+            return { error: true, data: error.message };
```
Comment (girish-thoughtspot) [inline src/embed/base.ts:167]: Its still not rejecting the promise, replace with `throw error`.
Same for exportTML function.

File: src/embed/base.ts [hunk 6]
```diff
@@ -109,6 +129,87 @@ export const prefetch = (url?: string, prefetchFeatures?: PrefetchFeatures[]): v
...
+
+    return fetch(`${thoughtSpotHost}${EndPoints.EXPORT_TML}`, {
+        method: 'POST',
+        headers,
+        body: JSON.stringify(payload),
+        credentials: 'include',
+    })
+        .then((response) => {
+            if (!response.ok) {
+                console.log(response);
```
Comment (girish-thoughtspot) [inline src/embed/base.ts:202]: Clear console log statements.

Comment (adityamittal3107) [reply src/embed/base.ts:202]: Done.

File: src/embed/base.ts (hunk 5 above)
Comment (adityamittal3107) [reply src/embed/base.ts:167]: Done.

**Review by girish-thoughtspot — APPROVED** [review]

**Pull request merged**